src/mame/drivers/leapfrog_iquest.cpp

```diff
@@ -0,0 +1,124 @@
+// license:BSD-3-Clause
+// copyright-holders:David Haywood
+/******************************************************************************
+
+   Leapfrog IQuest
+
+   has LCD display, resolution unknown
+
+*******************************************************************************/
+
+#include "emu.h"
+
+#include "cpu/mcs51/mcs51.h"
+
+#include "bus/generic/slot.h"
+#include "bus/generic/carts.h"
+
+#include "screen.h"
+#include "softlist.h"
+#include "speaker.h"
+#include "screen.h"
+
+class leapfrog_iquest_state : public driver_device
+{
+public:
+	leapfrog_iquest_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag)
+		, m_maincpu(*this, "maincpu")
+		, m_cart(*this, "cartslot")
+		, m_screen(*this, "screen")
+		, m_cart_region(nullptr)
+	{ }
+
+	void leapfrog_iquest(machine_config &config);
+
+private:
+	virtual void machine_start() override;
+	virtual void machine_reset() override;
+
+	void prog_map(address_map &map);
+	void ext_map(address_map &map);
+
+	uint32_t screen_update(screen_device &screen, bitmap_rgb32 &bitmap, const rectangle &cliprect);
+
+	DECLARE_DEVICE_IMAGE_LOAD_MEMBER(cart_load);
+
+	required_device<mcs51_cpu_device> m_maincpu;
+	required_device<generic_slot_device> m_cart;
+	required_device<screen_device> m_screen;
+	memory_region *m_cart_region;
+};
+
+
+
+void leapfrog_iquest_state::machine_start()
+{
+	// if there's a cart, override the standard mapping
+	if (m_cart && m_cart->exists())
+	{
+		std::string region_tag;
+		m_cart_region = memregion(region_tag.assign(m_cart->tag()).append(GENERIC_ROM_REGION_TAG).c_str());
```

Comment:
You can do `memregion(std::string(m_cart->tag()) + GENERIC_ROM_REGION_TAG)` rather than the voodoo with the local variable.  I think that pattern came from overly-literal translation of the old astring code to C++.

Reply:
i'll look into it, the driver was cloned from leapfrog_leappad at the moment because the CPU is the same, and the cart form factor not too different, although this is clearly a different system with a screen etc.